WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -606,48 +618,123 @@ - (UITableViewCell *)configureFeaturedImageCellForIndexPath:(NSIndexPath *)index
-
-        if ([[LocationService sharedService] locationServiceRunning]) {
-            [actCell.spinner startAnimating];
-            actCell.textLabel.text = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
-        } else {
-            actCell.textLabel.text = NSLocalizedString(@"Set Location", @"Geolocation feature to set the location.");
-            [actCell.spinner stopAnimating];
-        }
-
-        cell = actCell;
-
+- (void)showLocationError:(NSError *)error {
+    NSString *title = NSLocalizedString(@"Location", @"Title for alert when a generic error happened when trying to find the location of the device");
+    NSString *message = NSLocalizedString(@"There was a problem when trying to access your location. Please try again later.",  @"Explaining to the user there was an error trying to obtain the current location of the user.");
+    NSString *cancelText = NSLocalizedString(@"OK", "");
+    NSString *otherButtonTitle = nil;
+    if (error.domain == kCLErrorDomain && error.code == kCLErrorDenied) {
+        otherButtonTitle = NSLocalizedString(@"Open Settings", @"Go to the settings app");
+        title = NSLocalizedString(@"Location", @"Title for alert when access to the media library is not granted by the user");
+        message = NSLocalizedString(@"WordPress needs permission to access your device location in order to geotag your post. Please change the privacy settings if you wish to allow this.",  @"Explaining to the user why the app needs access to the device location.");
+    }
+    
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:cancelText style:UIAlertActionStyleCancel handler:nil];
+    [alertController addAction:okAction];
+    
+    if (otherButtonTitle) {
+        UIAlertAction *otherAction = [UIAlertAction actionWithTitle:otherButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
+            NSURL *settingsURL = [NSURL URLWithString:UIApplicationOpenSettingsURLString];
+            [[UIApplication sharedApplication] openURL:settingsURL];
+        }];
+        [alertController addAction:otherAction];
+    }
+    [self presentViewController:alertController animated:YES completion:nil];
+}
+
+- (void)updatePostLocationState:(BOOL)geoLocationOn {
+    if (geoLocationOn && self.locationService.locationServicesDisabled) {
+        [self showLocationError:[NSError errorWithDomain:kCLErrorDomain code:kCLErrorDenied userInfo:nil]];
+    } else if (geoLocationOn && !self.locationService.locationServicesDisabled) {
+        __weak __typeof__(self) weakSelf = self;
+        [self.locationService getCurrentLocationAndAddress:^(CLLocation *location, NSString *address, NSError *error) {
+            __typeof__(self) strongSelf = weakSelf;
```

Comment:
Sure.  My concern is how its constructed.  
First, a weak connection to self is defined `__weak __typeof__(self) weakSelf = self;` which will be used inside the block. No problem.  Second, the weak reference to self, `weakSelf`, is assigned to a new variable to get a strong reference: `__typeof__(self) strongSelf = weakSelf;`.  Also, no problem in principle but notice how its currently declared. Since `__typeof__(self)` is inside the block, a reference to `self` is being captured inside the block and will be retained. This defeats the purpose of the `weakSelf` reference.  If the goal is to have the weakSelf / strongSelf pattern, then strong self should be defined as `__typeof__(weakSelf) strongSelf = weakSelf;` rather than `__typeof__(self) strongSelf = weakSelf;`.